keyvalue/clients/src/main/java/tachyon/client/keyvalue/LinearProbingIndex.java

```diff
@@ -124,22 +133,34 @@ public boolean put(byte[] key, byte[] value, PayloadWriter writer) throws IOExce
 
   @Override
   public ByteBuffer get(ByteBuffer key, PayloadReader reader) {
+    int bucketOffset = bucketOffset(key, reader);
+    if (bucketOffset == - 1) {
+      return null;
+    }
+    return reader.getValue(ByteIOUtils.readInt(mBuf, bucketOffset + 1));
```

Comment:
nvm, forgot the fingerprint. Maybe we should have a utility method to go from buffer + bucketOffset to payload index